packages/mdl-menu/util.js

```diff
@@ -0,0 +1,140 @@
+// c1 and c2 are the matching coordinate on control points P1 and P2, respectively.
+// Control points P0 and P3 are assumed to be (0,0) and (1,1), respectively.
+// Adapted from https://github.com/google/closure-library/blob/master/closure/goog/math/bezier.js.
+function getBezierCoordinate_(t, c1, c2) {
+  // Special case start and end.
+  if (t === 0) {
+    return 0;
+  } else if (t === 1) {
+    return 1;
+  }
+
+  // Step one - from 4 points to 3
+  let ic0 = t * c1;
+  let ic1 = c1 + t * (c2 - c1);
+  const ic2 = c2 + t * (1 - c2);
+
+  // Step two - from 3 points to 2
+  ic0 += t * (ic1 - ic0);
+  ic1 += t * (ic2 - ic1);
+
+  // Final step - last point
+  return ic0 + t * (ic1 - ic0);
+}
+
+// Project a point onto the Bezier curve, from a given X. Calculates the position t along the curve.
+// Adapted from https://github.com/google/closure-library/blob/master/closure/goog/math/bezier.js.
+function solvePositionFromXValue_(xVal, x1, x2) {
+  // Desired precision on the computation.
+  const epsilon = 1e-6;
+
+  if (xVal <= 0) {
+    return 0;
+  } else if (xVal >= 1) {
+    return 1;
+  }
+
+  // Initial estimate of t using linear interpolation.
+  let t = xVal;
+
+  // Try gradient descent to solve for t. If it works, it is very fast.
+  let tMin = 0;
+  let tMax = 1;
+  let value = 0;
+  for (let i = 0; i < 8; i++) {
+    value = getBezierCoordinate_(t, x1, x2);
+    const derivative = (getBezierCoordinate_(t + epsilon, x1, x2) - value) / epsilon;
+    if (Math.abs(value - xVal) < epsilon) {
+      return t;
+    } else if (Math.abs(derivative) < epsilon) {
+      break;
+    } else {
+      if (value < xVal) {
+        tMin = t;
+      } else {
+        tMax = t;
+      }
+      t -= (value - xVal) / derivative;
+    }
+  }
+
+  // If the gradient descent got stuck in a local minimum, e.g. because
+  // the derivative was close to 0, use a Dichotomy refinement instead.
+  // We limit the number of interations to 8.
+  for (let i = 0; Math.abs(value - xVal) > epsilon && i < 8; i++) {
+    if (value < xVal) {
+      tMin = t;
+      t = (t + tMax) / 2;
+    } else {
+      tMax = t;
+      t = (t + tMin) / 2;
+    }
+    value = getBezierCoordinate_(t, x1, x2);
+  }
+  return t;
+}
+
+/**
+ * Returns the easing value to apply at time t, for a given cubic bezier curve.
+ * Control points P0 and P3 are assumed to be (0,0) and (1,1), respectively.
+ * @param {number} time The current time in the animation, scaled between 0 and 1.
+ * @param {number} x1 The x value of control point P1.
+ * @param {number} y1 The y value of control point P1.
+ * @param {number} x2 The x value of control point P2.
+ * @param {number} y2 The y value of control point P2.
+ * @return {number} The easing value to apply (between 0 and 1).
+ */
+function bezierProgress(time, x1, y1, x2, y2) {
+  return getBezierCoordinate_(solvePositionFromXValue_(time, x1, x2), y1, y2);
+}
+
+export {getTransformPropertyName, clamp, bezierProgress};
```

Comment:
rather than doing this, it would be less error-prone to just add an `export` statement in front of each function

```js
export function bezierProgress(time, x1, y1, x2, y2) { /* ... */ }
```